service/history/stateBuilder.go

```diff
@@ -695,29 +762,33 @@ func (b *stateBuilderImpl) scheduleDeleteHistoryTimerTask(event *shared.HistoryE
 	return b.getTimerBuilder(event).createDeleteHistoryEventTimerTask(time.Duration(retentionInDays) * time.Hour * 24), nil
 }
 
-func (b *stateBuilderImpl) scheduleUpsertSearchAttributesTask() persistence.Task {
-	return &persistence.UpsertWorkflowSearchAttributesTask{}
+func (b *stateBuilderImpl) appendTasksForFinishedExecutions(
+	event *shared.HistoryEvent,
+	domainID string,
+	workflowID string,
+) error {
+	b.transferTasks = append(b.transferTasks, b.scheduleDeleteHistoryTransferTask())
+	timerTask, err := b.scheduleDeleteHistoryTimerTask(event, domainID, workflowID)
+	if err != nil {
+		return err
+	}
+	b.timerTasks = append(b.timerTasks, timerTask)
+	return nil
 }
 
-func (b *stateBuilderImpl) getTaskList(msBuilder mutableState) string {
+func (b *stateBuilderImpl) getTaskList(
+	msBuilder mutableState,
+) string {
 	// on the standby side, sticky tasklist is meaningless, so always use the normal tasklist
```

Comment:
this comments means that only decision tasklist is used by passive processing logic.

sticky decision tasklist does not applies to passive side, since sticky decision ties to a specific worker, which exists at the active side